### PR TITLE
Pypy

### PR DESCRIPTION
**Pull request by mictadlo**

### CODE REVIEW

**Comment by mictadlo** [comment]
hi @conda-forge-coordinator ,
Any idea why it travis has failed?
Thank you in advance,
Michal

**Comment by mictadlo** [comment]
Hi @xoviat and @dougalsutherland,
Any idea why  travis has failed?
Thank you in advance,
Michal

File: recipes/pypy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+about:
+    home: http://pypy.org/
+    license: MIT
+    license_family: MIT
+    license_file: LICENSE
+    summary: PyPy is a Python interpreter and just-in-time compiler.
+
+extra:
+    final: true
```
Comment (CJ-Wright) [inline recipes/pypy/meta.yaml]: Is this needed?

File: recipes/pypy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  version: {{ version }}
+
+source:
+  url: https://bitbucket.org/squeaky/portable-pypy/downloads/pypy-5.10.0-linux_x86_64-portable.tar.bz2
+  fn: {{ name|lower }}_{{ version }}.tar.bz2
+  sha256: {{ hash }}
+
+
+build:
+    number: '0'
```
Comment (CJ-Wright) [inline recipes/pypy/meta.yaml]: I think this can be just `0`

File: recipes/pypy/meta.yaml [hunk 3]
```diff
@@ -14,21 +14,9 @@ source:
 
 build:
     number: '0'
+    skip: True # [not osx]
```
Comment (CJ-Wright) [inline recipes/pypy/meta.yaml]: This skips everything but OSX you want `skip: True  # [osx]`

File: recipes/pypy/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  sha256: {{ hash }}
+
+
+build:
+    number: '0'
+    skip: True  # [not osx]
+
+requirements:
+    run:
+        - openssl
```
Comment (dougalsutherland) [inline recipes/pypy/meta.yaml]: Some of these will probably need to be have version constraints, depending on how the upstream package is built. I'll investigate later.

**Comment by mictadlo** [comment]
Hi,
I used [skip: True  # [not osx]](https://github.com/mictadlo/staged-recipes/blob/dbe2e52450035ca1c8ef9ddf4eb3ed66a311c568/recipes/pypy/meta.yaml#L17) but Travis still tries to test it for [OSX](https://travis-ci.org/conda-forge/staged-recipes/builds/335000227#L236)

`TEST START: /Users/travis/miniconda/conda-bld/osx-64/pypy2-5.10.0-0.tar.bz2`

This package is only for Linux.

Thank you in advance.

Michal

**Comment by xoviat** [comment]
@mictadlo What does this program do?

```go
func build(os string) {
   if !(os == 'osx') {
       skipBuild();
   } else {
       runBuild();
   }
}
```

**Comment by mictadlo** [comment]
@xoviat Where would I need to your code? PyPy is an alternative implementation of the Python programming language which often runs faster than the standard implementation of Python, CPython. The reason PyPy is faster is because it is a just-in-time compiler while CPython is an interpreter

**Comment by xoviat** [comment]
You don't need to use this code. You just need to understand it.

**Comment by xoviat** [comment]
No it won't. It will skip the build when it isn't OSX.

**Comment by xoviat** [comment]
Hint: you improperly translated your mental objective into program instructions. `skip: True # [not osx]` does not achieve your stated goal.

**Comment by mictadlo** [comment]
Do I have to change it to `skip: True # [osx]`?

**Comment by xoviat** [comment]
I think it's `skip: True # [osx or win]`.

**Comment by mictadlo** [comment]
Thank you all. This time all checks have been passed.

**Comment by mictadlo** [comment]
Hi @conda-forge-admin @conda-forge-coordinator  @conda-forge-manager  @conda-forge-status,

How is it possible to make this package available?

Thank you in advance.

Best wishes,

Michal

**Comment by CJ-Wright** [comment]
I think you want @conda-forge/staged-recipes the others are bots.

**Comment by Juanlu001** [comment]
Or @conda-forge/admin @conda-forge/admins

**Comment by Juanlu001** [comment]
Sorry: @conda-forge/core

File: recipes/pypy/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{% set name = "pypy2.7" %}
+{% set version = "5.10.0" %}
+{% set hash = "c966124497ba8728654ce1161fa4c46b035ff30f289be70960f58292e5897cc8" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://bitbucket.org/squeaky/portable-pypy/downloads/pypy-5.10.0-linux_x86_64-portable.tar.bz2
```
Comment (patricksnape) [inline recipes/pypy/meta.yaml]: Shouldn't we really be building this from scratch rather than copying a pre-built source?

Comment (mariusvniekerk) [reply recipes/pypy/meta.yaml]: This is a prebuilt source

Comment (scopatz) [reply recipes/pypy/meta.yaml]: @patricksnape not necessarily.  It is less CI burden to go from the prebuilt source. Sometimes there could be linking problems with prebuilt source or too many bundled dependencies which is why we'd go with building ourselves.

**Comment by scopatz** [comment]
I feel like the recipe can be improved in the feedstock as needed.

File: recipes/pypy/meta.yaml (hunk 5 above)
Comment (ocefpaf) [reply recipes/pypy/meta.yaml]: I tried to build pypy from scratch once and it took more than an hour. I don't think our CIs can handle that at the moment.

File: recipes/pypy/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{% set name = "pypy2.7" %}
```
Comment (isuruf) [inline recipes/pypy/meta.yaml]: Why is the name `pypy2.7` instead of `pypy`?

Comment (scopatz) [reply recipes/pypy/meta.yaml]: https://pypy.org/download.html  This is their naming scheme, so that it indicates CPython comparability.

Comment (jakirkham) [reply recipes/pypy/meta.yaml]: Right they have their own versioning in addition to CPython compat versioning. We might consider a `pypy` metapackage to make things a little easier.

**Comment by mictadlo** [comment]
Thank you all for your help.

**Comment by jakirkham** [comment]
For those interested in PyPy compatibility with CPython, would encourage you to take a look at issue ( https://github.com/conda-forge/pypy2.7-feedstock/issues/1 ) where this is being discussed.

**Comment by Juanlu001** [comment]
@mictadlo do you intend to add PyPy3.5 as well?

**Comment by mictadlo** [comment]
Hi @Juanlu001, At the moment, I do not have any plans to do PYPY3. I asked @squeaky-pl whether he would be interested to do it.

Thank you in advance.

Best wishes,

Michal

**Pull request merged**